File: CS/KeyboardHandler/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.Portable.Input;
#region #usings
using DevExpress.Services;
using DevExpress.XtraRichEdit;
#endregion #usings

namespace KeyboardHandler {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
            labelControl1.Text = "CTRL-C, CTRL-INS, CTRL-X key combinations are blocked for the RichEdit control";
            richEditControl1.LoadDocument("sample_document.rtf", DocumentFormat.Rtf);
            SubstituteKeyboardService();
        }
#region #substservice
        // ...
        private void SubstituteKeyboardService() {
            IKeyboardHandlerService service = 
                richEditControl1.GetService<IKeyboardHandlerService>();
            MyKeyboardHandlerServiceWrapper wrapper = 
                new MyKeyboardHandlerServiceWrapper(service);
            richEditControl1.RemoveService(typeof(IKeyboardHandlerService));
            richEditControl1.AddService(typeof(IKeyboardHandlerService), wrapper);
        }
        //...
        #endregion #substservice
    }
    #region #myservice
    public class MyKeyboardHandlerServiceWrapper : KeyboardHandlerServiceWrapper {
        public MyKeyboardHandlerServiceWrapper(IKeyboardHandlerService service) 
            : base(service) { }

        public override void OnKeyDown(PortableKeyEventArgs e) {
            if(((e.KeyCode == PortableKeys.C) || (e.KeyCode == PortableKeys.X) || 
                (e.KeyCode == PortableKeys.Insert)) && e.Control) {
                return;
            }
            base.OnKeyDown(e);
        }
    }
    #endregion #myservice





}

File: VB/KeyboardHandler/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.Portable.Input
#Region "#usings"
Imports DevExpress.Services
Imports DevExpress.XtraRichEdit
#End Region ' #usings

Namespace KeyboardHandler
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()
			labelControl1.Text = "CTRL-C, CTRL-INS, CTRL-X key combinations are blocked for the RichEdit control"
			richEditControl1.LoadDocument("sample_document.rtf", DocumentFormat.Rtf)
			SubstituteKeyboardService()
		End Sub
#Region "#substservice"
		' ...
		Private Sub SubstituteKeyboardService()
			Dim service As IKeyboardHandlerService = richEditControl1.GetService(Of IKeyboardHandlerService)()
			Dim wrapper As New MyKeyboardHandlerServiceWrapper(service)
			richEditControl1.RemoveService(GetType(IKeyboardHandlerService))
			richEditControl1.AddService(GetType(IKeyboardHandlerService), wrapper)
		End Sub
		'...
		#End Region ' #substservice
	End Class
	#Region "#myservice"
	Public Class MyKeyboardHandlerServiceWrapper
		Inherits KeyboardHandlerServiceWrapper

		Public Sub New(ByVal service As IKeyboardHandlerService)
			MyBase.New(service)
		End Sub

		Public Overrides Sub OnKeyDown(ByVal e As PortableKeyEventArgs)
			If ((e.KeyCode = PortableKeys.C) OrElse (e.KeyCode = PortableKeys.X) OrElse (e.KeyCode = PortableKeys.Insert)) AndAlso e.Control Then
				Return
			End If
			MyBase.OnKeyDown(e)
		End Sub
	End Class
	#End Region ' #myservice





End Namespace